infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -149,9 +171,14 @@ public void onEntry(LogReplicationState from) {
             if (from != this) {
                 snapshotSender.reset();
                 fsm.getAckReader().markSnapshotSyncInfoOngoing(forcedSnapshotSync, transitionEventId);
-            }
+                snapshotSyncTimerSample = MeterRegistryProvider.getInstance().map(Timer::start);
 
-            transmitFuture = fsm.getLogReplicationFSMWorkers().submit(() -> snapshotSender.transmit(transitionEventId));
+            }
+            Runnable snapShotSendTask = () -> snapshotSender.transmit(transitionEventId);
+            if (senderTimer.isPresent()) {
+                snapShotSendTask = senderTimer.get().wrap(snapShotSendTask);
```

Comment:
I believe we would have the same issue described InLogEntrySyncState here.. the transmit will not reflect accurately the time for sending a snapshot message as it might not send anything at all...